solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -1149,7 +1150,7 @@ protected Query getFieldQuery(String field, List<String> queryTerms, boolean raw
         return newFieldQuery
             (getAnalyzer(), field, queryText, false, fieldAutoGenPhraseQueries, fieldEnableGraphQueries, synonymQueryStyle);
       } else {
-        if (raw) {
+        if (raw) {// assumption: raw = false only when called from ExtendedDismaxQueryParser.getQuery()
```

Comment:
I wonder whether it makes more sense to leave this "assumption" comment where it was -- at the point that the exception is swallowed? No doubt the comment is _true_ either place, but placing the comment up here separates it from where action is taken _based_ on that assumption.

Reply:
no strong opinion here, I thought it made more sense, on the side of the boolean variable it explains, I can revert if you think there are any benefit.
N.B.  we are not swallowing the exception anymore, so I am not sure it has any sense there anymore

Reply:
Yeah, either way's fine. I meant that we're "swallowing" the exception in the sense that we're not _rethrowing_ it, which lenient approach is warranted (iiuc) by the assumption that this code is only called from from the lenient edismax parser. I guess my thought was: if I'm a developer looking at the exception get caught and not rethrown, I'm likely to wonder "why be lenient and not rethrow? -- ah! because of this commented assumption!". By contrast, if I'm a developer looking at the {{if (raw) {}} line, I'll see the comment and wonder "why do I care about where I'm being called from at this point?".

Reply:
I see your point and I agree. Just reverted in the code!